assets/src/adaptivity/rules-engine.ts

```diff
@@ -304,11 +303,21 @@ export const getReferencedKeysInConditions = (conditions: any) => {
     ) {
       // value could have more than one reference inside it
       const exprs = extractAllExpressionsFromText(condition.value);
+      const expressions = condition.value.match(/{([^{^}]+)}/g);
       exprs.forEach((expr: string) => {
         if (expr.search(/app\.|variables\.|stage\.|session\./) !== -1) {
           references.add(expr);
         }
       });
+      expressions.forEach((expr: string) => {
```

Comment:
The only reason why added a separate code here so that it does not break any unknown use case that I am not able to think. I am sure this won't break anything.